src/utils/AuthApi.ts

```diff
@@ -0,0 +1,15 @@
+import { useAxios, ApiHookResponse } from './UseAxios';
+import { HttpMethods } from './HttpMethods';
+
+const baseUrl = process.env.REACT_APP_API_URL;
+
+export interface AuthenticationRequest {
+  user: {
```

Comment:
I think this abstraction would be stronger if the caller of `useAuthenticationAPI` doesn't have to construct the user object themselves but rather just gives the username and password when they call the hook.

Reply:
Passing the username and password as separate parameters when calling the hook would definitely make the abstraction stronger.

However, I defined an interface for all non-get requests because I want the client to explicitly label the values that they want to pass to the API. For example, with the `useAuthenticationRequest`, the client would have to label the username and password when making an authentication request. If username and password were passed as separate parameters, misordering the parameters would cause the request to be denied. Using the `AuthenticationRequest` interface makes some mistakes less likely.

Is it too overengineered?

Reply:
I thought about this more and I like your suggestion. I'll pass the username and password as parameters